include/xnnpack.h

```diff
@@ -273,8 +282,8 @@ enum xnn_datatype {
   /// 32-bit signed integers.
   xnn_datatype_int32 = 11,
   /// Quantized 4-bit signed integer with shared per-channel-block quantization
-  /// parameters.
-  xnn_datatype_qbint4 = 12,
+  /// parameters (bf16 format).
+  xnn_datatype_qbint4_bf16 = 12,
```

Comment:
How does this specialization trickle down to gemm-config and kernels selection? It might be OK for now since we don't have other d-types but imagine similar set of kernels but for fp16 scales. We need to disambiguate them at some level. 

This is also applicable to an extent to other APIs like create/reshape etc. They might need to know what is the scale dtype.